sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample12_ConsumeEventsWithEventProcessor.cs

```diff
@@ -35,17 +40,250 @@ public class Sample12_ConsumeEventsWithEventProcessor : IEventHubsSample
         public async Task RunAsync(string connectionString,
                                    string eventHubName)
         {
-            // We will start by creating a client using its default set of options.
+            // We will start by creating a client using its default set of options.  It will be used by the event processor to
+            // communicate with the Azure Event Hubs service.
 
             await using (var client = new EventHubClient(connectionString, eventHubName))
             {
-                //TODO (caio, pri1): Populate the sample
+                // An event processor is associated with a specific Event Hub and a consumer group.  It receives events from
+                // all existing partitions in the Event Hub, passing them to the user for processing.
```

Comment:
nit:  "existing" doesn't feel necessary since partitions can't be changed once after the EH is created.